cilium-dbg/cmd/statedb.go

```diff
@@ -27,7 +33,67 @@ var statedbDumpCmd = &cobra.Command{
 	},
 }
 
+func statedbTableCommand[Obj statedb.TabWritable](tableName string) *cobra.Command {
+	var watchInterval time.Duration
+	cmd := &cobra.Command{
+		Use:   tableName,
+		Short: fmt.Sprintf("Show contents of table %q", tableName),
+		Run: func(cmd *cobra.Command, args []string) {
+			table := statedb.NewRemoteTable[Obj](client, tableName)
+
+			w := tabwriter.NewWriter(os.Stdout, 5, 0, 3, ' ', 0)
+			var obj Obj
+			w.Write([]byte(obj.TabHeader()))
+			defer w.Flush()
+
+			revision := statedb.Revision(0)
+
+			for {
+				// Query the contents of the table by revision, so that objects
+				// that were last modified are shown last.
+				iter, errChan := table.LowerBound(context.Background(), statedb.ByRevision[Obj](revision))
```

Comment:
nit: LowerBound doesn't do it for me. Perhaps startsWith vs exactMatch??

Reply:
`LowerBound` is inherited to statedb from `go-memdb` and it's the terminology used by `immutable-radix`.

Reply:
Do wonder if API like this would make more sense:

```
type Table[Obj any] interface {
  Iterator(ReadTxn, Index) Iterator[Obj]
}

type Iterator[Obj] interface {
  SeekBeginning()
  SeekExact(Query)
  SeekGreaterOrEqual(Query)
  Next() (...)
}
```
e.g. closer to APIs of LevelDB and so on. Though does seem more fiddly to use to me.

Reply:
Thanks for the details Jussi! 
Another approach would be for the query object to dictate how to shape the iterator aka Seek(Query) while the query would be build to match the expectation of the user aka EQ, GTE, etc... Hence the matching logic would rest on the Query object and the iterator remains clean. This would allow us to also offer ordering semantics aka reverse counterparts  without bloating the iterator interface??

Reply:
As this discussion isn't about this PR, let's continue it on slack.